.pre-commit-config.yaml

```diff
@@ -4,36 +4,73 @@ ci:
 exclude: ".*\\.asdf$"
 
 repos:
+  # basic checks
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: v6.0.0
     hooks:
       - id: check-added-large-files
       - id: check-ast
       - id: check-case-conflict
       - id: check-yaml
-        args: ["--unsafe"]
+        args:
+          - --unsafe
       - id: check-toml
       - id: check-merge-conflict
       - id: check-symlinks
       - id: debug-statements
       - id: detect-private-key
       # - id: end-of-file-fixer
       # - id: trailing-whitespace
+  # simple text searches for common issues
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.10.0
+    hooks:
+      - id: rst-directive-colons
+      # - id: rst-inline-touching-normal
+      - id: text-unicode-replacement-char
+  # lint and format
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: 'v0.12.10'
+    rev: v0.12.10
     hooks:
-      - id: ruff
-        args: ["--fix"]
+      - id: ruff-check
+        args:
+          - --fix
+          - --show-fixes
       - id: ruff-format
+  # type checking
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.17.1
+    hooks:
+      - id: mypy
+  # spell-checking
   - repo: https://github.com/codespell-project/codespell
     rev: v2.4.1
     hooks:
-    - id: codespell
-      args: ["--write-changes", "--summary"]
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.17.1  # Use the latest stable version
+      - id: codespell
+        args:
+          - --write-changes
+          - --summary
+        additional_dependencies:
+          - tomli
+  # validate change log and fragments in `changes/`
+  - repo: https://github.com/twisted/towncrier
+    rev: 24.8.0
     hooks:
-      - id: mypy
+      - id: towncrier-check
+  # format YAML and Markdown
```

Comment:
I would prefer we don't leave commented out checks. We either use it or don't. Why add them but comment them out?

Reply:
These are checks used in other repos / dependencies that we should, ideally, aim to use eventually. We can gradually enable these in other PRs though